core/src/network_manager/worker.rs

```diff
@@ -113,12 +121,33 @@ impl Worker for NetworkManager {
     }
 
     fn start(&mut self) -> Result<(), Self::Error> {
-        // Build a transport and connect it to the P2P swarm
-        // TODO: Migrate this to QUIC
-        let transport = block_on(libp2p::development_transport(self.local_keypair.clone()))
-            .map_err(|err| NetworkManagerError::SetupError(err.to_string()))?;
+        // Build a quic transport with tcp fallback
+        let hostport = format!("/ip4/0.0.0.0/tcp/{}", self.config.port);
+        let addr: Multiaddr = hostport.parse().unwrap();
+
+        // Build the quic transport
+        let config = QuicConfig::new(&self.local_keypair);
+        let quic_transport = QuicTransport::new(config);
```

Comment:
This isn't really in scope for this PR, but [this issue](https://github.com/bluejekyll/trust-dns/issues/1946) meant we had to change our transport to test correctly. We've been meaning to migrate to QUIC once libp2p support was better anyways, so I added it to this PR.